app/controllers/application_controller.rb

```diff
@@ -5,6 +5,159 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "super_application_secret"
   end
 
+  get "/" do 
+    erb :index
+  end
+
+  get "/signup" do 
+    if logged_in?
+      redirect to "/tweets"
+    end
+    erb :"/users/create_user"
+  end
+
+  post "/signup" do 
+    if params[:username] == "" || params[:email] == "" || params[:password] == ""
+      redirect to "/signup"
+    else
+      @user = User.create(params)
+      session[:user_id] = @user.id # log in the user after they sign up
+      redirect to "/tweets"
+    end
+  end
+
+    get '/login' do 
+    if logged_in?
+      redirect '/tweets'
+    else
+      erb :'users/login'
+    end
+  end
+
+  post '/login' do
+    #raise params.inspect
+    @user = User.find_by(:username => params["username"])
+    if @user && @user.authenticate(params["password"])
+      session[:user_id] = @user.id
+      redirect "/tweets"
+    else
+      redirect to '/login' #or login?
+    end
+  end
+
+  get '/logout' do
+    if logged_in?
+      session.clear
+      redirect "/login"
+    else
+      redirect "/login"
+    end
+  end
+########################### tweets ###########################
+  get "/tweets" do 
+    #@user = current_user
+    @tweets = Tweet.all
+    if logged_in?
+      @user = current_user
+      erb :"/tweets/tweets"
+    else
+      redirect to "/login"
+    end
+  end
+
+  get '/tweets/new' do 
+    if logged_in?
+      erb :'/tweets/create_tweet'
+    else
+      redirect "/login"
+    end
+  end
+
+  post '/tweets' do
+  #raise params.inspect 
+    if params[:tweet][:content] == ""
+      redirect '/tweets/new'
+    else
+      @user = User.find_by_id(session[:user_id])
+      @tweet = Tweet.create(:content => params[:tweet][:content], :user_id => session[:user_id])
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
@tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.